### PR TITLE
Updated setup.py and added pyproject.toml

### PR DESCRIPTION
**Pull request by joseavegaa**
Updated the deprecated `setup.py` install and created the more standard `pyproject.toml`. It creates a wheel which is installable and works on `Python >=3.6`.

I have attached a ZIP file with the generated wheel and tar.gz from `python -m build`

Let me know if there is a problem, 

[dist.zip](https://github.com/danni/python-pkcs11/files/9743634/dist.zip)

### CODE REVIEW

**Comment by gregbreen** [comment]
@danni it would be very handy if you could spare some time for this PR.

File: dev-requirements.txt
Comment (danni) [inline dev-requirements.txt:1]: It seems like a shame to lose the generated version of this we had through pip-tools. Or at least to keep it as short as it was.

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+[build-system]
+requires = ["setuptools >= 67.8.0", 
+            "cython >= 0.29.35",
+            "asn1crypto >= 1.5.1",
```
Comment (danni) [inline pyproject.toml:4]: This has mixed build requirements with runtime requirements. Specifically there's two different versions of asn1crypto in play.

File: dev-requirements.txt
Comment (danni) [reply dev-requirements.txt:1]: i.e. if the intention is to drop pip-tools, then can we just copy this file to dev-requirements.txt

**Review by danni — COMMENTED** [review]
Hi yeah, sorry. PRs for this package occasionally fall off my radar.

**Review by kalebmckale — COMMENTED** [review]
A `requirements.txt` file is generally a pinned-version file for a specific installation, equivalent to a "lock" file. The version specifications are added directly to `setup.py` (or through file `requirements.in` like previously done) or to `pyproject.toml`. I personally prefer using `pipenv` but if one wants to have a secure "lock" file as @danni seems to intend, usually you'll want an external package like `pipenv` or `pip-tools` to do the dependency resolution.

**Pull request closed without merging**